django_x509/base/admin.py

```diff
@@ -55,15 +56,51 @@ def get_fields(self, request, obj=None):
         return fields
 
 
-class CaAdmin(BaseAdmin):
+class AbstractCaAdmin(BaseAdmin):
     list_filter = ['key_length', 'digest', 'created']
+    fields = ['operation_type',
+              'name',
+              'notes',
+              'key_length',
+              'digest',
+              'validity_start',
+              'validity_end',
+              'country_code',
+              'state',
+              'city',
+              'organization_name',
+              'email',
+              'common_name',
+              'extensions',
+              'serial_number',
+              'certificate',
+              'private_key',
+              'created',
+              'modified']
+
+    class Media:
+        js = ('django-x509/js/switcher.js',)
 
 
-class CertAdmin(BaseAdmin):
+class AbstractCertForm(forms.ModelForm):
+        OPERATION_CHOICES = (
+            ('-----', '-----'),
+            ('new', _('Create new')),
+            ('import', _('Import Existing'))
+
+        )
+        operation_type = forms.ChoiceField(choices=OPERATION_CHOICES)
+
+        class Meta:
+            fields = '__all__'
+
+
+class AbstractCertAdmin(BaseAdmin):
```

Comment:
I'm sorry.. we can't change this or dependencies will break, but we can add an alias to keep backward compatibility:

`CertAdmin = AbstractCertAdmin` same applies for `Ca`